apps/typescript-project-references-e2e/tests/typescript-project-references.spec.ts

```diff
@@ -0,0 +1,106 @@
+import {
+    ensureNxProject,
+    runNxCommandAsync,
+    runCommandAsync,
+    uniq,
+    cleanup,
+    updateFile,
+    readFile,
+} from '@nrwl/nx-plugin/testing'
+
+jest.setTimeout(200000)
+
+describe('typescript-project-references e2e', () => {
+    it('should create typescript-project-references', async () => {
```

Comment:
NX allows you to write end to end tests against your generator.

It creates a new NX project, then you can run commands against it and assert the state. Pretty nifty.

This way I can actually automate the conversion and write tests to make sure all the various things (like testing, builds, packaging npm packages etc) all work